server/src/domain/job/job.service.ts

```diff
@@ -163,9 +164,15 @@ export class JobService {
         await this.jobRepository.queue({ name: JobName.CLASSIFY_IMAGE, data: item.data });
         await this.jobRepository.queue({ name: JobName.ENCODE_CLIP, data: item.data });
         await this.jobRepository.queue({ name: JobName.RECOGNIZE_FACES, data: item.data });
+        if (item.data.source !== 'upload') {
+          break;
+        }
 
         const [asset] = await this.assetRepository.getByIds([item.data.id]);
         if (asset) {
+          if (asset.type === AssetType.VIDEO) {
```

Comment:
Can you give a more explanation on why moving the job here would help with race conditions?

Reply:
Currently metadata extraction and transcoding are queued at the same time. Since metadata extraction is followed by migration, this can cause transcoding to fail. But since thumbnail generation is queued only after migration, there's no longer any risk of a race condition.